HousingRepairsOnlineApi/Helpers/NotificationConfiguration/CommunalNotificationConfigurationProvider.cs

```diff
@@ -0,0 +1,78 @@
+{
+    public string ConfirmationSmsTemplateId { get; set; }
+    public string ConfirmationEmailTemplateId { get; set; }
+    public string InternalEmailTemplateId { get; set; }
+
+    public CommunalNotificationConfigurationProvider(string confirmationSmsTemplateId, string confirmationEmailTemplateId,
+        string internalEmailTemplateId)
+    {
+        this.ConfirmationSmsTemplateId = confirmationSmsTemplateId;
+        this.ConfirmationEmailTemplateId = confirmationEmailTemplateId;
+        this.InternalEmailTemplateId = internalEmailTemplateId;
+    }
+    public Dictionary<string, dynamic> GetPersonalisationForInternalEmailTemplate(Repair repair)
+    {
+        Guard.Against.NullOrWhiteSpace(repair.Id, nameof(repair.Id), "The repair reference provided is invalid");
+        Guard.Against.NullOrWhiteSpace(repair.Address.LocationId, nameof(repair.Address.LocationId), "The uprn provided is invalid");
+        Guard.Against.NullOrWhiteSpace(repair.Address.Display, nameof(repair.Address.Display), "The address provided is invalid");
+        Guard.Against.NullOrWhiteSpace(repair.SOR, nameof(repair.SOR), "The sor provided is invalid");
+        Guard.Against.NullOrWhiteSpace(repair.Description.Text, nameof(repair.Description.Text), "The repairDescription provided is invalid");
+        Guard.Against.NullOrWhiteSpace(repair.ContactDetails?.Value, nameof(repair.ContactDetails.Value), "The contact number provided is invalid");
+
+        return new Dictionary<string, dynamic>
+        {
+            {"repair_ref", repair.Id},
+            {"uprn", repair.Address.LocationId},
+            {"address", repair.Address.Display},
+            {"sor", repair.SOR},
+            {"repair_desc", repair.Description.Text},
+            {"contact_no", repair.ContactDetails?.Value}
+        };
+    }
+
+    public Dictionary<string, dynamic> GetPersonalisationForEmailTemplate(Repair repair)
+    {
+        Guard.Against.NullOrWhiteSpace(repair.Id, nameof(repair.Id), "The booking reference provided is invalid");
+        Guard.Against.NullOrWhiteSpace(repair.Time.Display, nameof(repair.Time.Display), "The appointment time provided is invalid");
+
+        return new Dictionary<string, dynamic>
+        {
+            {"repair_ref", repair.Id},
+            {"appointment_time", repair.Time.Display}
+        };
+    }
+
+    public Dictionary<string, dynamic> GetPersonalisationForSMSTemplate(Repair repair)
+    {
+        Guard.Against.NullOrWhiteSpace(repair.Id, nameof(repair.Id), "The booking reference provided is invalid");
+        Guard.Against.NullOrWhiteSpace(repair.Time.Display, nameof(repair.Time.Display), "The appointment time provided is invalid");
+
+        return new Dictionary<string, dynamic>
+        {
+            {"repair_ref", repair.Id},
+            {"appointment_time", repair.Time.Display}
+        };
+    }
+
+    public async Task<string> GetImageLink(IRetrieveImageLinkUseCase retrieveImageLinkUseCase, Repair repair)
```

Comment:
This method is repeated twice - once for [tenant](https://github.com/Newarkandsherwood/housing-repairs-online-api/pull/36/files#diff-40e41965529f7261733f92c99f151faafce70af296ec3e434c6d06ca985d3ca5R66) and here for [Communal](https://github.com/Newarkandsherwood/housing-repairs-online-api/pull/36/files#diff-b692aaee1da12c929fe22d63bfb2d0d6075a9d46fe90d5239198b1a17487695cR66) (and again in future for leaseholder too).

Is it possible to reduce this duplication?

Reply:
I've added a base class "BaseNotificationConfigurationProvider" to specify the common behaviour.